numpy/core/src/multiarray/buffer.h

```diff
@@ -4,7 +4,7 @@
 extern NPY_NO_EXPORT PyBufferProcs array_as_buffer;
 
 NPY_NO_EXPORT void
-_array_dealloc_buffer_info(PyArrayObject *self);
+_array_dealloc_buffer_info(PyObject *self);
```

Comment:
Should it still be called `_array_dealloc_buffer_info` as now it is used to deallocate `buffer_info` from other objects?